node/actors/control_api.go

```diff
@@ -1,26 +1,197 @@
 package actors
 
 import (
+	"errors"
+	"fmt"
+	"log/slog"
+
 	"ergo.services/ergo/act"
 	"ergo.services/ergo/gen"
+	"github.com/nats-io/nats.go"
+	"github.com/nats-io/nkeys"
 )
 
-func createControlAPIServer() gen.ProcessBehavior {
-	return &controlAPIServer{}
+const APIPrefix = "$NEX"
+
+func createControlAPI() gen.ProcessBehavior {
+	return &controlAPI{
+		subsz: make([]*nats.Subscription, 0),
+	}
 }
 
-type controlAPIServer struct {
+type controlAPI struct {
 	act.Actor
+
+	publicKey  string
+	publicXKey string
+	subsz      []*nats.Subscription
 }
 
-func (api *controlAPIServer) Init(args ...any) error {
-	api.Log().Info("Control API Server started")
+func (api *controlAPI) Init(args ...any) error {
```

Comment:
For all of these `Init` methods that take variable arguments, it would be safer and easier to maintain/evolve if a single struct was passed with the arguments rather than relying on them positionally. Only one argument would need to be type-checked and validation could be more flexible.

Reply:
This makes a lot of sense. Each actor can expose a struct that contains the init options it needs and then just get it from args[0]

Note that the args slice is how ergo inits stuff and there's a whole rabbit hole behind why. (It's related to failover restarts)

Reply:
I figured it was an interface requirement.

Reply:
Yep. Also, the Ergo runtime keeps a copy of those args in case an actor crashes and it needs to restart it. The constructors are never re-invoked, but the "factory" method is called and then the `Init` function is re-invoked with the original params.

Reply:
All actors on this branch with the exception of `internalNatsServer` now conform to this suggested single-argument `params`. Will add the same for the internal NATS actor as part of #390 after we merge this PR.